tap_salesforce/sync.py

```diff
@@ -121,6 +121,18 @@ def sync_stream(sf, catalog_entry, state):
                 sync_report(sf, catalog_entry, state, counter)
             singer.write_state(state)
         except RequestException as ex:
+            if ex.response is not None:
+                code, message = None, None
+                try: 
+                    resp_json = ex.response.json()
+                    code = resp_json['exceptionCode']
+                    message = resp_json['exceptionMessage']
+                except Exception:
+                    pass
+
+                if code is not None and message is not None:
+                    raise SymonException(f'Import failed with the following Salesforce error: (error code: {code}) {message}', 'salesforce.SalesforceApiError')
+
             raise Exception("{} Response: {}, (Stream: {})".format(
                 ex, ex.response.text, stream)) from ex
```

Comment:
maybe we can remove this line. If ex.response is none, it would error with `has no attribute text` with `ex.response.text`

Reply:
There could be a case where ex.response is not None but returned error doesn't have 'exceptionCode' or 'exceptionMessage'. In that case, we could see what the error was through this line. 
It seems like some endpoints return 'errorCode' and 'message' while others return 'exceptionCode' and 'exceptionMessage'. I'll take a look.

Reply:
response object formats observed:
`[{"errorCode":"NOT_FOUND","message":"The requested resource does not exist"}]`
`{"exceptionCode":"InvalidJob","exceptionMessage":"Unable to find object: Account"}`